containers/Ideas/IdeaCreate.js

```diff
@@ -22,34 +22,108 @@ import API from '../../BaseApi';
 
 export const IdeaCreate = props => {
 
-  const [error, setError] = useState();
-  const {moduleId, project} = props.route.params;
+  const {idea, project, editing} = props.route.params;
+  const moduleId = project.single_agenda_setting_module;
   const arrowLeftIcon = (
     <IconSLI name='arrow-left' size={22} />
   );
+  const ideaValidationSchema = yup.object().shape({
+    name: yup
+      .string()
+      .required('Idea title is Required'),
+    description: yup
+      .string()
+      .min(20, ({ min }) => `Description must be at least ${min} characters`)
+      .max(100, 'Description must be no longer then 100 characters')
+      .required('Description is required'),
+  });
+
+  const [error, setError] = useState();
+  const [clicked, setClicked] = useState(false);
+  const [categories, setCategories] = useState([]);
+  const [selectedCategory, setSelectedCategory] = useState('');
+  const [labels, setLabels] = useState([]);
+  const [labelsChecked, setLabelsChecked] = useState([]);
 
   useEffect(() => {
     if (error) {
       Alert.alert('An error occured', error, [{ text: 'Ok' }]);
     }
     else {
-      fetchLabelsAndCategories();
+      setLabelsAndCategories();
+    }
+  }, [error, project]);
+
+  const setLabelsAndCategories = async() => {
+    const module = await API.getModule(moduleId);
+
+    if (module.categories) {
+      // map property names to format needed for formik
+      setCategories(module.categories.map(category => ({value: category.id, label: category.name})));
+      if (editing) {
```

Comment:
I actually already changed the naming quite some times and thought editing was descriptive enough :) But if you had problems understanding the code because of that, I can also change it again?

Reply:
probably a matter of preference, but i am totally okay with `editing` as well. lets keep it